kyo-data/shared/src/main/scala/kyo/Result.scala

```diff
@@ -6,132 +6,94 @@ import scala.annotation.targetName
 import scala.util.Try
 import scala.util.control.NonFatal
 
-/** Represents a result that can be either a success, a failure or an unexpected panic.
+/** A type representing computations that can succeed, fail with an expected error, or fail with an unexpected exception. Result combines
+  * the features of `Either` (for expected errors) and `Try` (for unexpected exceptions) into a single type that can handle both cases while
+  * maintaining type safety.
+  *
+  * Result has the following possible states:
+  *   - `Success[A]`: Contains a successful value of type `A`
+  *   - `Error[E]`: Base type for both Failure and Panic
+  *     - `Failure[E]`: Represents expected errors of type `E`
+  *     - `Panic`: Represents unexpected runtime exceptions
+  *
+  * Result provides several groups of operations:
+  *   - Fold operations (`foldError`, `foldFailureOrThrow`, `foldAll`) for matching on the different states
+  *   - Map operations (`map`, `mapError`, `mapFailure`, `mapPanic`) for transforming specific states
+  *   - FlatMap operations (`flatMap`, `flatMapError`, `flatMapFailure`, `flatMapPanic`) for sequencing computations and handling errors
+  *   - Query operations (`isSuccess`, `isError`, `isFailure`, `isPanic`, `contains`, `exists`, `forall`)
+  *   - Conversion operations (`toEither`, `toMaybe`, `toTry`)
+  *
+  * The implementation supports handling specific error types through `SafeClassTag`, allowing precise error handling for union types. This
+  * enables handling specific failures while maintaining other error types in the result:
+  *
+  * {{{
+  * result.flatMapError[NetworkError] { e =>  // Handles only NetworkError
+  *   Result.succeed(fallbackValue)
+  * } // Other error types remain unchanged
+  * }}}
+  *
+  * Result uses an unboxed representation through Scala 3's opaque types, allowing the successful value to be stored directly without
+  * additional allocation. This optimization is particularly important for high-performance code paths where the success case is the most
+  * common outcome. Additionally, methods are marked as `inline` where function dispatch would introduce overhead, ensuring that operations
+  * like `map` and `flatMap` are optimized at compile-time and avoid allocations.
   *
   * @tparam E
-  *   The type of the error in case of failure
+  *   The type of expected errors that can occur
   * @tparam A
-  *   The type of the value in case of success
+  *   The type of the successful value
   */
 opaque type Result[+E, +A] >: (Success[A] | Error[E]) = Success[A] | Error[E]
 
-/** Companion object for Result type */
 object Result:
-    import internal.*
 
-    inline given [E, A](using inline ce: CanEqual[A, A]): CanEqual[Result[E, A], Result[E, A]] = CanEqual.derived
```

Comment:
I've moved these implicit evidences to the end of the file. No changes in the impl